### PR TITLE
feat: Add `GetSpecSchema` to the plugin spec

### PR DESCRIPTION
**Pull request by candiduslynx**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1 @@
+.idea/
```
Comment (disq) [inline .gitignore:1]: Maybe not start a .gitignore where here, where there's none

File: plugin/v3/plugin.proto [hunk 2]
```diff
@@ -41,6 +45,15 @@ message GetVersion {
   }
 }
 
+message GetSpecSchema {
+  message Request {}
+  message Response {
+    // Should be a valid JSON schema for the plugin own (nested) spec.
```
Comment (hermanschaaf) [inline plugin/v3/plugin.proto:51]:
```suggestion
    // Should be a valid JSON schema for the plugin's own (nested) spec.
```

File: plugin/v3/plugin.proto [hunk 3]
```diff
@@ -12,6 +12,10 @@ service Plugin {
   rpc GetName(GetName.Request) returns (GetName.Response);
   // Get the current version of the plugin
   rpc GetVersion(GetVersion.Request) returns (GetVersion.Response);
+  // Get plugin own (nested) spec schema.
```
Comment (hermanschaaf) [inline plugin/v3/plugin.proto:15]:
```suggestion
  // Get plugin's own (nested) spec schema.
```

File: .gitignore (hunk 1 above)
Comment (hermanschaaf) [reply .gitignore:1]: +1, you can add `.idea` to your global gitignore file

Comment (candiduslynx) [reply .gitignore]: 16e68541329ddc036ac22a31c967c4539b6fc5fb

File: plugin/v3/plugin.proto (hunk 2 above)
Comment (candiduslynx) [reply plugin/v3/plugin.proto:51]: I cast thy `attributive noun` meaning for `plugin spec` used here!

File: plugin/v3/plugin.proto (hunk 3 above)
Comment (candiduslynx) [reply plugin/v3/plugin.proto:15]: I cast thy `attributive noun` meaning for `plugin spec` used here!

Comment (candiduslynx) [reply plugin/v3/plugin.proto]: 9e8d07bb634c781c024e7a461f3d963919fbadfc

File: plugin/v3/plugin.proto (hunk 2 above)
Comment (candiduslynx) [reply plugin/v3/plugin.proto]: 9e8d07bb634c781c024e7a461f3d963919fbadfc

File: plugin/v3/plugin.proto [hunk 4]
```diff
@@ -41,6 +45,15 @@ message GetVersion {
   }
 }
 
+message GetSpecSchema {
+  message Request {}
+  message Response {
+    // Should be a valid JSON schema for the plugin spec.
+    // See https://json-schema.org for more details.
+    optional bytes json_schema = 1;
```
Comment (candiduslynx) [inline plugin/v3/plugin.proto:53]:
```suggestion
    optional string json_schema = 1;
```

**Review by yevgenypats — APPROVED** [review]

**Review by hermanschaaf — APPROVED** [review]

**Pull request merged**